akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/SnapshotMutableStateSpec.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2017-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.scaladsl
+
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+
+import akka.actor.testkit.typed.scaladsl._
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.persistence.SelectedSnapshot
+import akka.persistence.SnapshotMetadata
+import akka.persistence.SnapshotSelectionCriteria
+import akka.persistence.snapshot.SnapshotStore
+import akka.persistence.typed.PersistenceId
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import org.scalatest.WordSpecLike
+
+object SnapshotMutableStateSpec {
+
+  class SlowInMemorySnapshotStore extends SnapshotStore {
+
+    private var state = Map.empty[String, (Any, SnapshotMetadata)]
+
+    def loadAsync(persistenceId: String, criteria: SnapshotSelectionCriteria): Future[Option[SelectedSnapshot]] = {
+      Future.successful(state.get(persistenceId).map {
+        case (snap, meta) ⇒ SelectedSnapshot(meta, snap)
+      })
+    }
+
+    def saveAsync(metadata: SnapshotMetadata, snapshot: Any): Future[Unit] = {
+      val snapshotState = snapshot.asInstanceOf[MutableState]
+      val value1 = snapshotState.value
+      Thread.sleep(50)
```

Comment:
Is there another check that can be done vs use this?

Reply:
then it would have to pass in a CountDownLatch in the snapshot, but that seems complicated here since the purpose of the test is to exercise race conditions